modules/python/src2/gen2.py

```diff
@@ -768,8 +792,15 @@ def gen_code(self, codegen):
             # if the function/method has only 1 signature, then just put it
             code += all_code_variants[0]
         else:
-            # try to execute each signature
-            code += "    PyErr_Clear();\n\n".join(["    {\n" + v + "    }\n" for v in all_code_variants])
```

Comment:
`PyErr_Clear()` was lost?

Reply:
[PyErr_Fetch](https://docs.python.org/3/c-api/exceptions.html#c.PyErr_Fetch) retrieves and clears error indicator if it is set.